src/units/attack_type.cpp

```diff
@@ -572,6 +636,51 @@ bool attack_type::describe_modification(const config& cfg,std::string* descripti
 	return true;
 }
 
+attack_type::recursion_guard attack_type::update_variables_recursion() const
+{
+	// this shouldn't be const, but replacing the const-and-mutable mess in attack_type is a big task
```

Comment:
What do you mean it shouldn't be const? If it's only modifying `num_recursion_` then I think it's correct for it to be const.

Reply:
I'll remove the comment - it's a left-over from the tag-name based implementation.